.github/workflows/reusable-build-win.yml

```diff
@@ -0,0 +1,53 @@
+on:
+  workflow_call:
+    inputs:
+      free-threaded:
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  build_win_32:
+    name: 'build and test (x86)'
```

Comment:
@hugovk I would like to use the name as `build and test` variations for macOS and Linux too.
It looks more clear when we look at the CI status bar.
But I will do it as a separate PR.